### ISSUE TITLE
Decommission operation timeout is too small for DB cluster with enabled tablets feature

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylla-cluster-tests#8855]
### Issue description

Nemesis `disrupt_decommission_streaming_err` times out in the test with enabled tablets:
```
2024-09-21 07:32:46.414: (ClusterHealthValidatorEvent Severity.CRITICAL) period_type=one-time \
    event_id=2d16b283-0ee6-4651-9862-d60efd70c2d3: type=NodeStatus \
    node=longevity-large-partitions-200k-pks-db-node-b866c292-0-2 \
    error=Current node Node longevity-large-partitions-200k-pks-db-node-b866c292-0-2 [34.148.211.120 | 10.142.0.128]. \
    Node Node longevity-large-partitions-200k-pks-db-node-b866c292-0-5 [35.185.64.144 | 10.142.0.139] \
    (DecommissionStreamingErr nemesis target node) status is UL
``` 

Argus:
![Screenshot from 2024-09-26 22-22-11](https://github.com/user-attachments/assets/a2cb62bf-9843-4055-9deb-5e9ead454cff)

Looking at the cluster state everything was going ok all that time while timeout was not reached.

So, I increased the timeout for it and ran another test here: [scylla-staging/valerii/vp-longevity-large-partition-200k-pks-4days-gce-test#4](https://argus.scylladb.com/test/0547d448-263a-43a3-807a-28d23cde8836/runs?additionalRuns[]=44df7273-31de-43fb-a74d-9e506a3fe378)

After timeout increase the nemesis passed:
![Screenshot from 2024-09-26 22-23-17](https://github.com/user-attachments/assets/cc9c6612-c68b-49fe-867b-4e6cf60471ad)

Note that in the passed variant the `add node operation after decommission` was started at `11:39`.
So, decommission under heavy `write load` took about `2h15m` whereas current timeout is `1h20m`.

#### Steps to Reproduce

1. Setup a DB cluster with tablets
2. Run heavy write load with large partitions
3. Run `disrupt_decommission_streaming_err` nemesis

**Expected behavior:** SCT waits proper amount of time

**Actual behavior:** SCT raises timeout error too early

### Impact

False negative

### How frequently does it reproduce?

`100%`

### Installation details

SCT Version: master
Scylla version (or git commit hash): `6.3.0~dev-20240920.488a372fdc2c with build-id 4b4271c7ab8f30e2cf9bccc6b2bff818f4d64600`

### Logs
* test_id: b866c292-32ca-4145-8ec5-910a55c1d92f
* job log: [scylla-master/tier1/longevity-large-partition-200k-pks-4days-gce-test#36](https://argus.scylladb.com/test/996501ad-c137-40ef-8dcc-a7d12f0ba3b6/runs?additionalRuns[]=b866c292-32ca-4145-8ec5-910a55c1d92f)

**Comment by fruch** [comment]
@aleksbykov 

isn't that something you already identified ? during this case ? has been any work related to fixed that  ?

**Comment by timtimb0t** [comment]
the issue reproduced:
## Packages

Scylla version: `6.3.0~dev-20241101.19a43b58599c` with build-id `8db6e96d95afbd2686d3ceaa32521f5696ac3b23`


Kernel Version: `6.8.0-1017-gcp`


## Installation details

Cluster size: 5 nodes (n2-highmem-16)

Scylla Nodes used in this run:
- longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 (35.237.109.129 | 10.142.0.6) (shards: 14)
- longevity-large-partitions-200k-pks-db-node-718b6c14-0-5 (34.139.37.228 | 10.142.0.172) (shards: 14)
- longevity-large-partitions-200k-pks-db-node-718b6c14-0-4 (35.231.9.235 | 10.142.0.111) (shards: 14)
- longevity-large-partitions-200k-pks-db-node-718b6c14-0-3 (34.139.169.109 | 10.142.0.11) (shards: 14)
- longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 (34.74.189.42 | 10.142.0.166) (shards: 14)
- longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 (34.74.51.135 | 10.142.0.155) (shards: -1)


OS / Image: `https://www.googleapis.com/compute/v1/projects/scylla-images/global/images/scylla-6-3-0-dev-x86-64-2024-11-02t01-59-26` (gce: undefined_region)

Test: `longevity-large-partition-200k-pks-4days-gce-test`
Test id: `718b6c14-43d1-412a-bf6d-375ca3f2b10a`
Test name: `scylla-master/tier1/longevity-large-partition-200k-pks-4days-gce-test`
Test method: `longevity_large_partition_test.LargePartitionLongevityTest.test_large_partition_longevity`
Test config file(s):

- [longevity-large-partition-200k_pks-4days.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/ba373698898e0c53406abaecd82ac311fa012f25/test-cases/longevity/longevity-large-partition-200k_pks-4days.yaml)


<details>
<summary>
Logs and commands
</summary>


- Restore Monitor Stack command: `$ hydra investigate show-monitor 718b6c14-43d1-412a-bf6d-375ca3f2b10a`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=718b6c14-43d1-412a-bf6d-375ca3f2b10a)
- Show all stored logs command: `$ hydra investigate show-logs 718b6c14-43d1-412a-bf6d-375ca3f2b10a`


## Logs:
- **longevity-large-partitions-200k-pks-db-node-718b6c14-0-4** - [https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_041105/longevity-large-partitions-200k-pks-db-node-718b6c14-0-4-718b6c14.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_041105/longevity-large-partitions-200k-pks-db-node-718b6c14-0-4-718b6c14.tar.gz)
- **db-cluster-718b6c14.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/db-cluster-718b6c14.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/db-cluster-718b6c14.tar.gz)
- **sct-runner-events-718b6c14.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/sct-runner-events-718b6c14.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/sct-runner-events-718b6c14.tar.gz)
- **sct-718b6c14.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/sct-718b6c14.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/sct-718b6c14.log.tar.gz)
- **loader-set-718b6c14.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/loader-set-718b6c14.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/loader-set-718b6c14.tar.gz)
- **monitor-set-718b6c14.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/monitor-set-718b6c14.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/718b6c14-43d1-412a-bf6d-375ca3f2b10a/20241102_111652/monitor-set-718b6c14.tar.gz)


[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/tier1/job/longevity-large-partition-200k-pks-4days-gce-test/42/)
[Argus](https://argus.scylladb.com/test/996501ad-c137-40ef-8dcc-a7d12f0ba3b6/runs?additionalRuns[]=718b6c14-43d1-412a-bf6d-375ca3f2b10a)
</details>

**Comment by fruch** [comment]
@aleksbykov and @soyacz 

call to `http://127.0.0.1:10000/storage_service/quiesce_topology` timeout in 1h, it probably should be using adaptive timeout and have bigger number then 1h

**Comment by fruch** [comment]
or it's a scylla issue, that need to be chased.

other way we should ignore it

**Comment by roydahan** [comment]
should or shouldn't?

**Comment by roydahan** [comment]
@aleksbykov / @temichus please check with raft team if there is an agreement of what should be the timeout for tablets to be balanced.
We can have it as adaptive timeout that will not fail it completely and will raise an event if we crossed the expected timeout.
But we will start collecting history of how long it should take.

**Comment by bhalevy** [comment]
@vponomaryov what version was it (please make a habit to document that in the issue description)
We need to open a scylla/scylla-enterprise issue for that.

@denesb / @tgrabiec please look into this and see why decommission is so much slower with tablets.

**Comment by tgrabiec** [comment]
Looking at db-cluster-718b6c14.

Decommission nemesis start:
```
2024-11-02 08:29:28.365: (DisruptionEvent Severity.NORMAL) period_type=begin event_id=1d3458c8-b229-430c-9ab1-1da702490613: nemesis_name=DecommissionStreamingErr target_node=Node longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 [34.74.189.42 | 10.142.0.166]
```

Server confirms promptly:
```
Nov 02 08:29:30.218408 longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 scylla[4494]:  [shard  0:strm] api - decommission
Nov 02 08:29:30.222493 longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 scylla[4494]:  [shard  0:strm] raft_topology - request decommission for: 44296656-966f-444e-aa7f-eb71d048428f
Nov 02 08:29:30.323016 longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 scylla[4494]:  [shard  0:strm] raft_topology - decommission: waiting for completion (request ID: 9479c080-98f4-11ef-53d7-c0983027b0be)
```

Failure reported 2h40m later:
```
2024-11-02 11:09:41.989: (DisruptionEvent Severity.ERROR) period_type=end event_id=1d3458c8-b229-430c-9ab1-1da702490613 duration=2h40m13s: nemesis_name=DecommissionStreamingErr target_node=Node longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 [34.74.189.42 | 10.14
2.0.166] errors=Wait for: Waiting decommission is finished for longevity-large-partitions-200k-pks-db-node-718b6c14-0-2...: timeout - 4800 seconds - expired
```

Tablet draining starts here, already 1h after decommission was initiated:
```
Nov 02 09:28:49.681958 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] raft_topology - entered `tablet draining` transition state
Nov 02 09:28:49.682012 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Examining DC us-east1 (shuffle=false, balancing=true)
Nov 02 09:28:49.682042 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Will drain node 44296656-966f-444e-aa7f-eb71d048428f (being_decommissioned) from DC us-east1
  Nov 02 09:28:49.682105 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Node 44296656-966f-444e-aa7f-eb71d048428f: rack=d avg_load=10.071428571428571, tablets=141, shards=14, state=being_decommissioned
Nov 02 09:28:49.682111 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Node 0881ead9-2b27-45cc-b5f5-4df41a12e78d: rack=d avg_load=10.071428571428571, tablets=141, shards=14, state=normal
Nov 02 09:28:49.682117 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Node 0cf767b3-e3c0-4bf6-870d-1836113e2e77: rack=d avg_load=10.071428571428571, tablets=141, shards=14, state=normal
Nov 02 09:28:49.682122 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Node 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0: rack=d avg_load=10, tablets=140, shards=14, state=normal
Nov 02 09:28:49.682126 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Node c61776f2-f94d-4289-8703-4b6878f1ce8a: rack=d avg_load=10.071428571428571, tablets=141, shards=14, state=normal
Nov 02 09:28:49.682602 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - target node: 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0, avg_load: 10, max: 10.071428571428571
```

What's blocking decommission is tablet migration, mostly intra-node migration:
```
Nov 02 08:23:03.562995 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:48 to 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:0
Nov 02 08:23:03.563026 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:43 to 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:7
Nov 02 08:23:03.563241 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:42 to 44296656-966f-444e-aa7f-eb71d048428f:10
Nov 02 08:23:03.563254 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:39 to c61776f2-f94d-4289-8703-4b6878f1ce8a:0
Nov 02 08:23:03.563262 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:37 to 44296656-966f-444e-aa7f-eb71d048428f:7
Nov 02 08:23:03.563270 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:36 to 0cf767b3-e3c0-4bf6-870d-1836113e2e77:9
Nov 02 08:23:03.563279 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:33 to 0cf767b3-e3c0-4bf6-870d-1836113e2e77:4
Nov 02 08:23:03.563287 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:45 to 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:1
Nov 02 08:23:03.563295 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:32 to 44296656-966f-444e-aa7f-eb71d048428f:9
Nov 02 08:23:03.563303 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:25 to 0881ead9-2b27-45cc-b5f5-4df41a12e78d:9
Nov 02 08:23:03.563312 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:44 to c61776f2-f94d-4289-8703-4b6878f1ce8a:13
Nov 02 08:23:03.563319 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:18 to c61776f2-f94d-4289-8703-4b6878f1ce8a:10
Nov 02 08:23:03.563326 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:49 to 0cf767b3-e3c0-4bf6-870d-1836113e2e77:2
Nov 02 08:23:03.563334 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:51 to 0881ead9-2b27-45cc-b5f5-4df41a12e78d:10
Nov 02 08:23:03.563353 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:53 to 0881ead9-2b27-45cc-b5f5-4df41a12e78d:9
Nov 02 08:23:03.563362 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:55 to 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:13
Nov 02 08:23:03.563369 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:58 to 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:3
Nov 02 08:23:03.563382 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:59 to 0cf767b3-e3c0-4bf6-870d-1836113e2e77:8
Nov 02 08:23:03.563390 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:34 to 0881ead9-2b27-45cc-b5f5-4df41a12e78d:12
Nov 02 08:23:03.563397 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet streaming (intranode_migration) of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:63 to 44296656-966f-444e-aa7f-eb71d048428f:6
Nov 02 08:31:55.987525 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:49 on 0cf767b3-e3c0-4bf6-870d-1836113e2e77:6
Nov 02 08:32:47.556655 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:37 on 44296656-966f-444e-aa7f-eb71d048428f:2
Nov 02 08:34:50.717842 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:39 on c61776f2-f94d-4289-8703-4b6878f1ce8a:2
Nov 02 08:35:05.512746 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:51 on 0881ead9-2b27-45cc-b5f5-4df41a12e78d:5
Nov 02 08:35:37.266871 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:59 on 0cf767b3-e3c0-4bf6-870d-1836113e2e77:10
Nov 02 08:36:41.507112 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:43 on 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:4
Nov 02 08:40:26.033827 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:32 on 44296656-966f-444e-aa7f-eb71d048428f:0
Nov 02 08:40:52.270961 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:44 on c61776f2-f94d-4289-8703-4b6878f1ce8a:4
Nov 02 08:42:08.753484 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:55 on 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:8
Nov 02 08:43:11.850867 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:53 on 0881ead9-2b27-45cc-b5f5-4df41a12e78d:11
Nov 02 08:43:46.265256 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:33 on 0cf767b3-e3c0-4bf6-870d-1836113e2e77:0
Nov 02 08:46:21.575995 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:25 on c61776f2-f94d-4289-8703-4b6878f1ce8a:11
Nov 02 08:46:30.883720 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:63 on 44296656-966f-444e-aa7f-eb71d048428f:11
Nov 02 08:49:38.390334 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:58 on 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:10
Nov 02 08:50:36.786546 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:18 on c61776f2-f94d-4289-8703-4b6878f1ce8a:9
Nov 02 08:51:24.427539 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:36 on 0cf767b3-e3c0-4bf6-870d-1836113e2e77:1
Nov 02 08:55:08.307149 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:42 on 44296656-966f-444e-aa7f-eb71d048428f:4
Nov 02 08:56:51.987210 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:34 on 0881ead9-2b27-45cc-b5f5-4df41a12e78d:4
Nov 02 08:58:09.377096 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:48 on 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:6


Nov 02 09:28:44.295531 node-0-1 scylla[4495]:  [shard  0: gms] raft_topology - Initiating tablet cleanup of 96dfa140-98dc-11ef-a830-0f9f45ed3c51:45 on 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:5
```

In particular, intra-node migration of tablet 96dfa140-98dc-11ef-a830-0f9f45ed3c51:45 to 1ac81849-0a4d-4cfa-9397-2e91ee3d83f0:5 (node 5):
```
Nov 02 08:23:03.658788 longevity-large-partitions-200k-pks-db-node-718b6c14-0-5 scylla[4494]:  [shard  0:strm] raft_topology - Starting intra-node streaming of tablet 96dfa140-98dc-11ef-a830-0f9f45ed3c51:45 from shard 5 to 1

Nov 02 09:28:39.769778 longevity-large-partitions-200k-pks-db-node-718b6c14-0-5 scylla[4494]:  [shard  0:strm] raft_topology - Finished intra-node streaming of tablet 96dfa140-98dc-11ef-a830-0f9f45ed3c51:45 from shard 5 to 1
```

It takes 1 hour, probably because it is blocked on split compactions. 

The intra-node migrations are due to imbalance caused by earlier split, here:
```
Nov 02 08:20:38.450705 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Table 96dfa140-98dc-11ef-a830-0f9f45ed3c51 with tablet_count=32 has an average tablet size of 46470187888
Nov 02 08:20:38.450711 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Table 8d14d990-98d3-11ef-9835-f7dd5d46f7ae with tablet_count=32 has an average tablet size of 5852051
Nov 02 08:20:38.450716 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Table 034ca210-98d3-11ef-9835-f7dd5d46f7ae with tablet_count=128 has an average tablet size of 10738467583
Nov 02 08:20:38.450723 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Finalizing resize decision for table 96dfa140-98dc-11ef-a830-0f9f45ed3c51 as all replicas agree on sequence number 3
Nov 02 08:20:38.450729 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Prepared 1 migration plans, out of which there were 0 tablet migration(s) and 1 resize decision(s)
Nov 02 08:20:38.450752 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Split tablets for table 96dfa140-98dc-11ef-a830-0f9f45ed3c51, increasing tablet count from 32 to 64
Nov 02 08:20:38.451193 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] raft_topology - updating topology state: Finished tablet split finalization
``` 

The table is still above split threshold, so load balancer immediately puts it into split mode (avg tablet size is 23 GB): 

```
Nov 02 08:21:45.729202 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Table 034ca210-98d3-11ef-9835-f7dd5d46f7ae with tablet_count=128 has an average tablet size of 10738467583
Nov 02 08:21:45.729209 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Table 8d14d990-98d3-11ef-9835-f7dd5d46f7ae with tablet_count=32 has an average tablet size of 5852051
Nov 02 08:21:45.729214 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Table 96dfa140-98dc-11ef-a830-0f9f45ed3c51 with tablet_count=64 has an average tablet size of 23235093944
Nov 02 08:21:45.729227 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Emitting resize decision of type split for table 96dfa140-98dc-11ef-a830-0f9f45ed3c51 due to avg tablet size of 23235093944
```
 
It doesn't finish, so it runs concurrently with decommission. 

The decommission node is drained in 02:43:43

```
Nov 02 11:14:54.700138 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard  0: gms] load_balancer - Node 44296656-966f-444e-aa7f-eb71d048428f is already drained, ignoring
```

**Comment by tgrabiec** [comment]
I can see 1 mutation error during this decommission event (3 in the whole run) in the logs, which shouldn't be there (source node is the decommissioned node):
```
2024-11-02T08:41:03.072+00:00 longevity-large-partitions-200k-pks-db-node-718b6c14-0-3  !WARNING | scylla[4489]:  [shard 10:stmt] storage_proxy - Stale topology detected, request has been fenced out, local fence version 695, request topology version 694, caller address 10.142.0.166

2024-11-02T08:41:03.072+00:00 longevity-large-partitions-200k-pks-db-node-718b6c14-0-3  !WARNING | scylla[4489]:  [shard 10:stmt] storage_proxy - Failed to apply mutation from 10.142.0.166#10: replica::stale_topology_exception (stale topology exception, caller version 694, callee fence version 695)
```

There are barrier stalls, up to 141 s, but not during decommission, where they are at the level of 2 s:

```
$ grep "held for" */system.log | clmn 3 4 15 | sort -n -k3 -r | head -n 60

08:23:00.422671 longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 141.809
05:50:53.308638 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 98.957
05:50:53.100661 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 98.749
05:42:38.130267 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 81.444
05:42:37.919981 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 81.234
08:21:45.865016 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 67.355
05:47:51.280709 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 48.938
05:41:16.424994 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 48.919
05:47:50.941562 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 48.599
05:41:15.998481 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 48.492
05:44:35.383737 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 47.844
05:44:35.279272 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 47.739
08:21:24.853091 longevity-large-partitions-200k-pks-db-node-718b6c14-0-3 46.240
05:53:24.986667 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 45.069
05:53:24.984037 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 45.066
05:56:44.092680 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 44.562
05:56:43.962613 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 44.432
05:51:36.668941 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 43.075
05:51:36.518678 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 42.925
05:49:14.175561 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 41.976
08:21:20.568487 longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 41.955
05:49:14.017318 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 41.817
05:48:31.924346 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 40.368
05:48:31.766232 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 40.210
08:21:18.347084 longevity-large-partitions-200k-pks-db-node-718b6c14-0-3 39.734
05:45:14.818958 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 39.079
08:21:15.844500 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 37.334
08:21:15.108080 longevity-large-partitions-200k-pks-db-node-718b6c14-0-3 36.495
08:21:14.884288 longevity-large-partitions-200k-pks-db-node-718b6c14-0-2 36.270
05:43:47.308743 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 36.177
05:43:47.062814 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 35.931
05:48:24.985547 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 33.430
05:48:24.952671 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 33.397
08:21:11.608632 longevity-large-partitions-200k-pks-db-node-718b6c14-0-5 32.995
05:55:09.140806 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 30.935
05:55:08.767274 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 30.561
05:52:39.595847 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 28.710
05:52:39.359190 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 28.473
05:57:12.003884 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 27.468
05:57:12.000406 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 27.464
05:55:35.470991 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 25.927
05:55:35.461101 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 25.917
08:21:04.180440 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 25.670
05:54:05.485088 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 24.871
05:54:05.479609 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 24.865
05:55:59.273317 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 23.583
05:55:59.259491 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 23.569
08:21:01.149042 longevity-large-partitions-200k-pks-db-node-718b6c14-0-5 22.536
05:47:02.163665 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 20.748
05:47:02.161700 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 20.746
05:47:22.582114 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 20.239
05:47:22.565712 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 20.223
08:20:57.587307 longevity-large-partitions-200k-pks-db-node-718b6c14-0-3 18.974
05:43:10.893136 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 18.594
05:43:10.278794 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 17.980
08:20:55.225514 longevity-large-partitions-200k-pks-db-node-718b6c14-0-5 16.612
05:53:40.416514 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 15.202
05:53:40.271106 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 15.057
05:40:27.303237 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 14.359
05:40:26.671270 longevity-large-partitions-200k-pks-db-node-718b6c14-0-6 13.725
``` 

Nov 02 08:21:45.865016 longevity-large-partitions-200k-pks-db-node-718b6c14-0-1 scylla[4495]:  [shard 12:stmt] token_metadata - topology version 652 held for 67.355 [s] past expiry, released at: 

```
[Backtrace #0]
void seastar::backtrace<seastar::current_backtrace_tasklocal()::$_0>(seastar::current_backtrace_tasklocal()::$_0&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:68
 (inlined by) seastar::current_backtrace_tasklocal() at ./build/release/seastar/./build/release/seastar/./seastar/src/util/backtrace.cc:97
seastar::current_tasktrace() at ./build/release/seastar/./build/release/seastar/./seastar/src/util/backtrace.cc:148
seastar::current_backtrace() at ./build/release/seastar/./build/release/seastar/./seastar/src/util/backtrace.cc:181
locator::token_metadata_impl::~token_metadata_impl() at ././locator/token_metadata.cc:1237
std::default_delete<locator::token_metadata_impl>::operator()(locator::token_metadata_impl*) const at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/unique_ptr.h:93
 (inlined by) std::unique_ptr<locator::token_metadata_impl, std::default_delete<locator::token_metadata_impl> >::~unique_ptr() at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/unique_ptr.h:398
 (inlined by) locator::token_metadata::~token_metadata() at ././locator/token_metadata.cc:909
seastar::lw_shared_ptr_no_esft<locator::token_metadata>::~lw_shared_ptr_no_esft() at ././seastar/include/seastar/core/shared_ptr.hh:170
 (inlined by) seastar::internal::lw_shared_ptr_accessors_no_esft<locator::token_metadata>::dispose(seastar::lw_shared_ptr_counter_base*) at ././seastar/include/seastar/core/shared_ptr.hh:226
 (inlined by) seastar::lw_shared_ptr<locator::token_metadata const>::~lw_shared_ptr() at ././seastar/include/seastar/core/shared_ptr.hh:324
 (inlined by) locator::effective_replication_map::~effective_replication_map() at ././locator/abstract_replication_strategy.hh:196
 (inlined by) locator::tablet_effective_replication_map::~tablet_effective_replication_map() at ././locator/tablets.cc:684
 (inlined by) seastar::shared_ptr_count_for<locator::tablet_effective_replication_map>::~shared_ptr_count_for() at ././seastar/include/seastar/core/shared_ptr.hh:485
 (inlined by) seastar::shared_ptr_count_for<locator::tablet_effective_replication_map>::~shared_ptr_count_for() at ././seastar/include/seastar/core/shared_ptr.hh:485
seastar::shared_ptr<locator::effective_replication_map const>::~shared_ptr() at ././seastar/include/seastar/core/shared_ptr.hh:563
 (inlined by) service::abstract_read_executor::~abstract_read_executor() at ././service/storage_proxy.cc:5097
service::never_speculating_read_executor::~never_speculating_read_executor() at ././service/storage_proxy.cc:5504
seastar::shared_ptr<service::abstract_read_executor>::~shared_ptr() at ././seastar/include/seastar/core/shared_ptr.hh:563
 (inlined by) std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >::~pair() at /usr/include/boost/container/detail/std_fwd.hpp:39
 (inlined by) void std::destroy_at<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> > >(std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_construct.h:88
 (inlined by) void std::_Destroy<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> > >(std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_construct.h:149
 (inlined by) void std::_Destroy_aux<false>::__destroy<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*>(std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*, std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_construct.h:163
 (inlined by) void std::_Destroy<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*>(std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*, std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_construct.h:195
 (inlined by) void std::destroy<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*>(std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*, std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >*) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/stl_construct.h:253
 (inlined by) utils::small_vector<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >, 1ul>::clear() at ././utils/small_vector.hh:268
 (inlined by) utils::small_vector<std::pair<seastar::shared_ptr<service::abstract_read_executor>, interval<dht::token> >, 1ul>::~small_vector() at ././utils/small_vector.hh:248
 (inlined by) service::storage_proxy::query_singular(seastar::lw_shared_ptr<query::read_command>, std::vector<interval<dht::ring_position>, std::allocator<interval<dht::ring_position> > >&&, db::consistency_level, service::storage_proxy_coordinator_query_options) at ././service/storage_proxy.cc:5856
std::__n4861::coroutine_handle<seastar::internal::coroutine_traits_base<boost::outcome_v2::basic_result<service::storage_proxy_coordinator_query_result, utils::exception_container<exceptions::mutation_write_timeout_exception, exceptions::read_timeout_exception, exceptions::read_failure_exception, exceptions::rate_limit_exception, exceptions::overloaded_exception>, utils::exception_container_throw_policy> >::promise_type>::resume() const at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/coroutine:242
 (inlined by) seastar::internal::coroutine_traits_base<boost::outcome_v2::basic_result<service::storage_proxy_coordinator_query_result, utils::exception_container<exceptions::mutation_write_timeout_exception, exceptions::read_timeout_exception, exceptions::read_failure_exception, exceptions::rate_limit_exception, exceptions::overloaded_exception>, utils::exception_container_throw_policy> >::promise_type::run_and_dispose() at ././seastar/include/seastar/core/coroutine.hh:80
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:2599
seastar::reactor::run_some_tasks() at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3067
seastar::reactor::do_run() at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:3235
std::_Function_handler<void (), seastar::smp::configure(seastar::smp_options const&, seastar::reactor_options const&)::$_0>::_M_invoke(std::_Any_data const&) at ./build/release/seastar/./build/release/seastar/./seastar/src/core/reactor.cc:4552
seastar::posix_thread::start_routine(void*) at /usr/lib/gcc/x86_64-redhat-linux/14/../../../../include/c++/14/bits/std_function.h:591
```

**Comment by aleksbykov** [comment]
1. PR: https://github.com/scylladb/scylla-cluster-tests/pull/8843 should fix this issue. and correctly wait that decommission is done
2. Rerun job. Reproduce issue
3. Fix timeout for decommission process

**Comment by aleksbykov** [comment]
Start reproducer: https://jenkins.scylladb.com/job/scylla-staging/job/abykov/job/longevity-large-partition-200k-pks-4days-gce-test/

**Comment by mykaul** [comment]
@aleksbykov - any updates?

**Comment by aleksbykov** [comment]
Increasing decommission timeout help to discover another issue: https://github.com/scylladb/scylladb/issues/22496
After several runs and checking the logs next was found:
if decommission was aborted at moment when tablets streaming is started, then the node after reboot return backed to cluster, and command rebuild is stucked. More details in issue: https://github.com/scylladb/scylladb/issues/22496
The problem reproduced only on gce for now

**Comment by yarongilor** [comment]
> Increasing decommission timeout help to discover another issue: [scylladb/scylladb#22496](https://github.com/scylladb/scylladb/issues/22496) After several runs and checking the logs next was found: if decommission was aborted at moment when tablets streaming is started, then the node after reboot return backed to cluster, and command rebuild is stucked. More details in issue: [scylladb/scylladb#22496](https://github.com/scylladb/scylladb/issues/22496) The problem reproduced only on gce for now

i wonder if a rebuild issue could be related to https://github.com/scylladb/scylladb/issues/17575
